src/modules/fancyzones/dll/dllmain.cpp

```diff
@@ -143,7 +152,20 @@ class FancyZonesModule : public PowertoyModuleIface
     {
         app_name = GET_RESOURCE_STRING(IDS_FANCYZONES);
         m_settings = MakeFancyZonesSettings(reinterpret_cast<HINSTANCE>(&__ImageBase), FancyZonesModule::get_name());
-        FancyZonesDataInstance().LoadFancyZonesData();
+        try
+        {
+            FancyZonesDataInstance().LoadFancyZonesData();
+        }
+        catch (...)
+        {
+            // Not being able to find path to persisted data completely breaks down FancyZones functionality.
+            std::wstring errorMessage = GET_RESOURCE_STRING(IDS_FANCYZONES_DATA_ERROR) + L" " + NonLocalizable::PowerToysIssuesURL;
+            MessageBox(NULL,
+                       errorMessage.c_str(),
+                       GET_RESOURCE_STRING(IDS_POWERTOYS_FANCYZONES).c_str(),
+                       MB_OK | MB_ICONERROR);
+            disableCore = true;

```

Comment:
Is this a fatal error? What about using default data?

Reply:
Hm, You're right, now that we have priority grid with 3 zones in it, we could do most of the regular fancyzones functionalities. I'll investigate to see amount of changes needed to have this in place. I assume there will be a need for a lot of if-else statements based on current implementation.

Reply:
What I meant is: if loading data fails, we should not disabled FZ but just ignore/reset the corrupted data.

Reply:
In this case it is not that data within json file is wrong (we handle that properly), it is the fact that we couldn't open the file in the first place. Therefore we cannot load any data, but maybe we can overwrite it with new data, if we retry opening the file again.

Reply:
We should sleep for 1 or 2 seconds and retry, and only after the second failure we should either disable the module or decide to nuke the file and start fresh.
For now let's add a retry and then report the error and disable the module. We don't know if this is happening frequently and if nuking the file is the best approach.

Reply:
Personally, I think that chance of this happening is quite low, but sure, I'll add a retry after a 1sec sleep. Later we can proceed with nuking the file.